clients/unshaded/src/main/java/tachyon/client/file/FileSystemMasterClient.java

```diff
@@ -161,69 +141,55 @@ public Void call() throws TachyonTException, TException {
   /**
    * Deletes a file or a directory.
    *
-   * @param id the id
-   * @param recursive whether to delete the file recursively (when it is a directory)
-   * @return whether operation succeeded or not
+   * @param path the path to delete
+   * @param options method options
    * @throws IOException if an I/O error occurs
    * @throws TachyonException if a Tachyon error occurs
    */
-  public synchronized boolean delete(final long id, final boolean recursive)
+  public synchronized void delete(final TachyonURI path, final DeleteOptions options)
       throws IOException, TachyonException {
-    return retryRPC(new RpcCallableThrowsTachyonTException<Boolean>() {
+    retryRPC(new RpcCallableThrowsTachyonTException<Void>() {
       @Override
-      public Boolean call() throws TachyonTException, TException {
-        return mClient.remove(id, recursive);
+      public Void call() throws TachyonTException, TException {
+        // TODO(calvin): Look into changing the master side implementation to take a uri
```

Comment:
Why not do it in this PR? The only change is to add a uri structure in thrift, right?